dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/ErrorDecoder.java

```diff
@@ -45,32 +59,58 @@ public enum ErrorDecoder {
 
     private static final SafeLogger log = SafeLoggerFactory.get(ErrorDecoder.class);
     private static final ObjectMapper MAPPER = ObjectMappers.newClientObjectMapper();
-    private static final EndpointErrorDecoder<?> ENDPOINT_ERROR_DECODER =
-            new EndpointErrorDecoder<>(Collections.emptyMap());
 
     public boolean isError(Response response) {
-        return ENDPOINT_ERROR_DECODER.isError(response);
+        return 300 <= response.code() && response.code() <= 599;
     }
 
     public RuntimeException decode(Response response) {
         if (log.isDebugEnabled()) {
-            log.debug("Received an error response", EndpointErrorDecoder.diagnosticArgs(response));
+            log.debug("Received an error response", diagnosticArgs(response));
         }
         RuntimeException result = decodeInternal(response);
-        result.addSuppressed(EndpointErrorDecoder.diagnostic(response));
+        result.addSuppressed(diagnostic(response));
         return result;
     }
 
     private RuntimeException decodeInternal(Response response) {
         // TODO(rfink): What about HTTP/101 switching protocols?
         // TODO(rfink): What about HEAD requests?
 
-        Optional<RuntimeException> maybeQosException = ENDPOINT_ERROR_DECODER.checkCode(response);
-        if (maybeQosException.isPresent()) {
-            return maybeQosException.get();
+        int code = response.code();
+        switch (code) {
+            case 308 -> {
```

Comment:
I used the arrow-style switch case here because of a build failure from [the StatementSwitchToExpressionSwitch check](https://errorprone.info/bugpattern/StatementSwitchToExpressionSwitch)